packages/vite/src/node/config.ts

```diff
@@ -889,7 +889,7 @@ async function loadConfigFromBundledFile(
   const extension = path.extname(fileName)
   const defaultLoader = require.extensions[extension]!
   require.extensions[extension] = (module: NodeModule, filename: string) => {
-    if (filename === fileName) {
+    if (filename === fs.realpathSync(fileName)) {
```

Comment:
@patak-js Wasn't there something about `realpathSync` that it is bad? Cause it is slow or something?

Reply:
I don't know, we are not using it in the codebase. But we are using its async version `realPath` here: https://github.com/vitejs/vite/blob/0693d038e096dd4db3140507269c6aaeefd2186e/packages/vite/src/node/server/sourcemap.ts#L25, so I don't think performance is an issue here. It is only while reading the config file. But I let others more used to linked files to check this PR.

Reply:
I think `fs.realpathSync(fileName)` will be called many times, if it is here and `require` exists inside the config.
How about caching it like this?
```ts
const realFileName = fs.realpathSync(fileName)
require.extensions[extension] = (module: NodeModule, filename: string) => {
  if (filename === realFileName) {
```

Reply:
That makes sense @sapphi-red, would you like to send a commit with that change to this PR directly?

Reply:
I updated 👍